beout/terminal.py

```diff
@@ -313,14 +314,20 @@ def box(self, text):
            '\n' + style('└' + ('─' * (len(stripped) + 2)) + '┘'))
 
     def _timestamp(self):
-        return ''.join([
-            self._config.timestamp_bracket_style('['),
-            self._config.timestamp_style(datetime.now().strftime(self._config.timestamp_format)),
-            self._config.timestamp_bracket_style(']')
-        ])
+        if self._config.timestamp_enabled:
+            return ''.join([
+                self._config.timestamp_bracket_style('['),
+                self._config.timestamp_style(datetime.now().strftime(self._config.timestamp_format)),
+                self._config.timestamp_bracket_style(']')
+            ])
+        else:
+            return ''
 
     def line(self, text):
-        return self._timestamp() + ' ' + self._substeps.str() + text
+        timestamp = self._timestamp()
+        if timestamp != "":
```

Comment:
Doing it this way makes the `timestamp_enabled` config var kinda redundant - I can just set the format to the empty string, and the leading space goes away as well. Since "There should be one-- and preferably only one --obvious way to do it.", I'd rather have the "do we need an extra space?" logic in `_timestamp`.

Reply:
@ghoranyi Yearly ping.